lib/src/Base/IterativeStat/IterativeMoments.cxx

```diff
@@ -0,0 +1,294 @@
+      << " order max=" << orderMax_
+      << " centered moments=" << centeredMoments_.__repr__();
+  return oss;
+}
+
+/** String converter */
+String IterativeMoments::__str__(const String & offset) const
+{
+  return getCenteredMoments().__str__(offset);
+}
+
+/**  Accessor to the centered moments */
+Sample IterativeMoments::getCenteredMoments() const
+{
+  return centeredMoments_;
+}
+
+/** Accessor to the maximum order declared */
+UnsignedInteger IterativeMoments::getOrder() const
+{
+  return orderMax_;
+}
+
+/** Accessor to the mean */
+Point IterativeMoments::getMean() const
+{
+  if (iteration_ >= 1)
+  {
+    return centeredMoments_[0] / iteration_;
+  }
+  else
+  {
+    return centeredMoments_[0];
+  }
+}
+
+/** Accessor to the variance */
+Point IterativeMoments::getVariance() const
+{
+  if (!(iteration_ > 0)) throw InternalException(HERE) << "Error: cannot compute the variance per component of an empty sample.";
+  if (!(orderMax_ >= 2)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum should be at least 2.";
+
+  /* Special case for a size 1 */
+  if (iteration_ == 1) return Point(dimension_, 0.0);
+
+  /* unbiased estimator */
+  return centeredMoments_[1] / (iteration_ - 1);
+}
+
+/** Accessor to the skewness */
+Point IterativeMoments::getSkewness() const
+{
+  if (!(iteration_ >= 2)) throw InternalException(HERE) << "Error: cannot compute the skewness per component of a sample of size less than 2.";
+  if (!(orderMax_ >= 3)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum order should be at least 3.";
+
+  /* Special case for a size 2 */
+  if (iteration_ == 2) return Point(dimension_, 0.0);
+
+
+  Point result(dimension_);
+  const Point varianceEstimator(getVariance());
+  /* unbiased estimator */
+  for(UnsignedInteger d = 0; d < dimension_; ++d)
+  {
+    result[d] = iteration_ / ((iteration_ - 1.0) * (iteration_ - 2.0)) * centeredMoments_(2, d) / std::pow(varianceEstimator[d], 1.5) ;
+  }
+  return result;
+}
+
+/** Accessor to the kurtosis */
+Point IterativeMoments::getKurtosis() const
+{
+  if (!(iteration_ >= 3)) throw InternalException(HERE) << "Error: cannot compute the kurtosis per component of a sample of size less than 3.";
+  if (!(orderMax_ >= 4)) throw InternalException(HERE) << "Error: unavailable method, the declared maximum order is lower than 4.";
+
+  /* Special case for a size 3 */
+  if (iteration_ == 3) return Point(dimension_, 0.0);
+
+
+  Point result(dimension_);
+  const Point varianceEstimator(getVariance());
+  const Point skewnessEstimator(getSkewness());
+  const UnsignedInteger n = iteration_;
+
+  const Scalar factor1 = n * (n + 1.0) / ((n - 1.0) * (n - 2.0) * (n - 3.0));
+  /*
+  Unbiased estimator:
+  const Scalar factor2 = -3.0 * (n - 1.0) * (n - 1.0) / ((n - 2.0) * (n - 3.0));
+  */
+  /* Corrected with Fischer -3 factor */
+  const Scalar factor2 = -3.0 * (3.0 * n - 5) / ((n - 2.0) * (n - 3.0));
```

Comment:
This cannot be correct, because factor2 must converge to -3 when n goes to infinity.

Reply:
See https://github.com/openturns/openturns/issues/1929 if it helps.

Reply:
Thank you very much.

Reply:
For your information, I added a detailed explanation of the equation in #1929.